### PR TITLE
Create Projects and add tasks to projects

### PR DESCRIPTION
**Pull request by collinsugwu**
**What this PR does**

1. Pre-define some projects and store in local storage.
2. create new projects
3. create a task and assign it to a project
4. update and delete a task
5. check a task as completed or incompleted

### CODE REVIEW

File: src/tasks.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+  const validateInput = validate(titleVal, descVal, dateVal);
+
+  if (validateInput) {
+    document.querySelector('.bg-modal').style.display = 'none';
+
+    const newTodo = toDo(titleVal, descVal, dateVal, priorityVal, projectId);
+
+    saveTodo(newTodo);
+  }
```
Comment (maelfosso) [inline src/tasks.js:82]: DOM-Manipulation and Logic are in separate modules

File: src/tasks.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  refreshPage();
+}
+
+function deleteToDo(index) {
+  document.querySelector(`.delete-${index}`).addEventListener('click', function deleteButton() {
+    const todoId = this.getAttribute('data-id');
+    document.querySelector(`.todo-${todoId}`).style.display = 'none';
+    deleteTask(todoId);
+  });
+}
```
Comment (maelfosso) [inline src/tasks.js]: DOM-Manipulation and Logic are in separate modules

File: src/tasks.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  const select = document.createElement('SELECT');
+  select.setAttribute('class', 'custom-select');
+  select.setAttribute('id', 'select-project');
+  const ps = ProjectStorage().getProject();
+  ps.forEach((element, index) => {
+    const option = document.createElement('OPTION');
+    option.setAttribute('value', `${index}`);
+    option.innerHTML = `${element.name}`;
+    select.appendChild(option);
+  });
```
Comment (maelfosso) [inline src/tasks.js]: DOM-Manipulation and Logic are in separate modules

File: src/tasks.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        <td data-column="done"><a data-id="${index}" data-done="${element.done}" class="btn btn-primary done-${index} done">Done</a></td>
+      `;
+        document.querySelector('tbody').appendChild(tr);
+        doneState(index);
+        deleteToDo(index);
+        editToDo(index);
+      }
+    });
+  }
+}
```
Comment (maelfosso) [inline src/tasks.js]: DOM-Manipulation and Logic are in separate modules

File: src/project.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  const ps = ProjectStorage().getProject();
+  const ol = document.createElement('OL');
+  ol.setAttribute('class', 'project');
+  ps.forEach((element, index) => {
+    const li = document.createElement('LI');
+    li.innerHTML = `<a href="#" class="project-${index}" data-id="${index}">${element.name}</a>`;
+    ol.appendChild(li);
+  });
+  projectDiv.appendChild(ol);
+}());
```
Comment (maelfosso) [inline src/project.js]: DOM-Manipulation and Logic are in separate modules

File: src/project.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  }
+};
+
+const validate = (name) => {
+  if (name === '') {
+    document.querySelector('.error-message').innerHTML = 'Name cannot be blank';
+  } else {
+    saveProject(name);
+  }
+};
```
Comment (maelfosso) [inline src/project.js]: DOM-Manipulation and Logic are in separate modules

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Wonderful result :+1: 
However, there are some requirements that need to be met

1. As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/modules_requirement.png), DOM-Manipulation and Logic are in separate modules. Only from the `tasks.js` file, we can see both DOM-Manipulation and Logic together, in the `addTodo()` function for example. For other cases, please, take a look at the comment below.

To fix that, please, get back to the `SOLID` principles explained [here](https://www.theodinproject.com/courses/javascript/lessons/oop-principles) and (http://aspiringcraftsman.com/2011/12/08/solid-javascript-single-responsibility-principle/)

2. ToDo created doesn't appear. 
Screenshot from 2020-03-02 18-45-18
As you could see in the screenshot up there, there are todos in LocalStorage but nothing in the webpage.

3. Update your README. The link for the live redirect me to a 404 Page

Kindly make the suggested changes and submit a new code review request.
Thanks.

Mael FOSSO

File: src/tasks.js (hunk 1 above)
Comment (Oghenebrume50) [reply src/tasks.js:82]: @collinsugwu @abruzy This was not attended to, perhaps it was an oversight

File: src/project.js (hunk 6 above)
Comment (Oghenebrume50) [reply src/project.js]: @collinsugwu @abruzy  This was not attended to, how about creating a function that listens to the project title input field and passes the content to validate then displays that message only if it is blank

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED 

Hi @collinsugwu @abruzy  good job guys, this project is coming along really fine, although I noticed that todos are not being created under selected projects, so if a create a new project and select it then create a todo, it usually goes into the first project. This happens in development, in the production link it almost look random. But users should be able to specifically create todos under any desired project :slightly_smiling_face:  

Please look into that and request a re-review :slightly_smiling_face:, feel free to reach me on slack if there is any issue

**RAPHAEL NORIODE** :sunglasses:

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @collinsugwu , @abruzy 

Nice, please you have to debug one aspect mentioned in previous review
### when user selects a specific project, and clicks on `"Add a ToDo"`, after filling todo-form. the information goes to a random project (not the one selected). This has to be consistent

Please fix that

**Comment by collinsugwu** [comment]
> ## hello @collinsugwu , @abruzy
> Nice, please you have to debug one aspect mentioned in previous review
> 
> ### when user selects a specific project, and clicks on `"Add a ToDo"`, after filling todo-form. the information goes to a random project (not the one selected). This has to be consistent
> Please fix that

I have gone through the app and I'm yet to have that experience

**Review by MauricioRobayo — APPROVED** [review]
Well done!

Could not reproduce the bug. Tested carefully but all tasks were correctly assigned to the project I chose. 

The issue might be related to the UI.

At first, I thought that if I selected a project from the list and then created a todo, the todo would be assigned to the project I chose before on the list. But when the form to create a todo pops up, it asks again for the project, so we have to select there the project we want the todo to go, no matter which project we selected before on the list of todos on the left.

Since that's the only remaining issue and I had no success reproducing it, It's a thumbs up from me. I'll appreciate any correction if I missed something.

Status: __Approved__.

Optional: Improve the UI so once a project is selected from the list of projects on the left all todos created will default to that project. It is confusing otherwise.

**Pull request merged**